data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetAttestations.java

```diff
@@ -0,0 +1,79 @@
+
+import static tech.pegasys.teku.beaconrestapi.CacheControlUtils.CACHE_NONE;
+import static tech.pegasys.teku.beaconrestapi.RestApiConstants.COMMITTEE_INDEX;
+import static tech.pegasys.teku.beaconrestapi.RestApiConstants.RES_INTERNAL_ERROR;
+import static tech.pegasys.teku.beaconrestapi.RestApiConstants.RES_OK;
+import static tech.pegasys.teku.beaconrestapi.RestApiConstants.SLOT;
+import static tech.pegasys.teku.beaconrestapi.RestApiConstants.TAG_V1_BEACON;
+import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_BAD_REQUEST;
+
+import io.javalin.core.util.Header;
+import io.javalin.http.Context;
+import io.javalin.plugin.openapi.annotations.HttpMethod;
+import io.javalin.plugin.openapi.annotations.OpenApi;
+import io.javalin.plugin.openapi.annotations.OpenApiContent;
+import io.javalin.plugin.openapi.annotations.OpenApiResponse;
+import java.util.Optional;
+import tech.pegasys.teku.api.DataProvider;
+import tech.pegasys.teku.api.NodeDataProvider;
+import tech.pegasys.teku.api.response.v1.beacon.GetAttestationsResponse;
+import tech.pegasys.teku.beaconrestapi.SingleQueryParameterUtils;
+import tech.pegasys.teku.beaconrestapi.handlers.AbstractHandler;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.provider.JsonProvider;
+
+public class GetAttestations extends AbstractHandler {
+  public static final String ROUTE = "/eth/v1/beacon/pool/attestations";
+  private final NodeDataProvider nodeDataProvider;
+
+  public GetAttestations(final DataProvider dataProvider, final JsonProvider jsonProvider) {
+    super(jsonProvider);
+    this.nodeDataProvider = dataProvider.getNodeDataProvider();
+  }
+
+  @OpenApi(
+      path = ROUTE,
+      method = HttpMethod.GET,
+      summary = "Get attestations from operation pool",
+      tags = {TAG_V1_BEACON},
+      description =
+          "Retrieves attestations known by the node but not necessarily incorporated into any block.",
+      responses = {
+        @OpenApiResponse(
+            status = RES_OK,
+            content = @OpenApiContent(from = GetAttestationsResponse.class)),
+        @OpenApiResponse(status = RES_INTERNAL_ERROR),
+      })
+  @Override
+  public void handle(final Context ctx) throws Exception {
+    try {
+      Optional<UInt64> maybeSlot =
+          SingleQueryParameterUtils.getParameterValueAsUInt64IfPresent(ctx.queryParamMap(), SLOT);
+      Optional<UInt64> maybeCommitteeIndex =
+          SingleQueryParameterUtils.getParameterValueAsUInt64IfPresent(
+              ctx.queryParamMap(), COMMITTEE_INDEX);
+      ctx.header(Header.CACHE_CONTROL, CACHE_NONE);
+      ctx.result(
+          jsonProvider.objectToJSON(
+              new GetAttestationsResponse(
+                  nodeDataProvider.getAttestations(maybeSlot, maybeCommitteeIndex))));
+    } catch (IllegalArgumentException e) {
+      ctx.result("The slot or committee index could not be parsed.");
+      ctx.status(SC_BAD_REQUEST);
+    }
```

Comment:
I've created a BadRequestException which has a generic handler, makes life easier - just need to catch the appropritate places and throw the BadRequestException (getParameterValueAsUInt64IfPresent will throw it once i merge #3112)